### DISCUSSION TITLE
Looking for feedback on a proposal to help the Python Launcher find pyenv interpreters

### DISCUSSION THREAD

**Discussion by brettcannon** [pyenv/pyenv#2472] [Q&A]
In https://github.com/brettcannon/python-launcher/discussions/168 I have an idea on a plug-in solution for the Python Launcher for Unix (and VS Code) to let people more easily customize how the Launcher finds Python interpreters and environments. In pyenv's case I can see someone creating a locator as part of this to tell the Python Launcher about all of the Python interpreters pyenv has installed.

At this stage I'm just looking for feedback on the idea and the JSON payload that a locator would use. So if pyenv as a project has any feedback to provide I would love to hear it! To be clear, I have no specific expectations of work or code to be written by pyenv; I just want to make sure what I have seems reasonable from your perspective.

**Comment by native-api** [comment]
In my experience, https://peps.python.org/pep-0394 does the job of providing access to installed tools better than `py`:

* it provides access to other installed executables as well -- while in Windows, I have to run stuff like `py -3 -m pip`
* it's seamless and shebang-compatible -- I can run Python the same regardless of my configuration.

So I don't really see what benefits `py` would bring to UNIX.

**Reply by brettcannon** [reply]
You can also look at this as wanting feedback on the behalf of VS Code then as we are looking at implementing this proposal to help in finding Python interpreters installed on the machine no matter what people's workflow or environment/interpreter management tool-of-choice is.
